js/src/relay/WalletLinkRelay.ts

```diff
@@ -593,10 +591,36 @@ export class WalletLinkRelay extends WalletLinkRelayAbstract {
         }
 
         WalletLinkRelay.accountRequestCallbackIds.add(id)
-      } else if (
-        request.method === Web3Method.switchEthereumChain ||
-        request.method === Web3Method.addEthereumChain
-      ) {
+      } else if (request.method === Web3Method.addEthereumChain) {
+        const cancel = () => {
+          this.handleWeb3ResponseMessage(
+            Web3ResponseMessage({
+              id,
+              response: AddEthereumChainResponse({ isApproved: false, rpcUrl: "" })
+            })
+          )
+        }
+        const approve = (rpcUrl: string) => {
+          this.handleWeb3ResponseMessage(
+            Web3ResponseMessage({
+              id,
+              response: AddEthereumChainResponse({ isApproved: true, rpcUrl })
+            })
+          )
+        }
+
+        this.ui.addEthereumChain({
+          onCancel: cancel,
+          onApprove: approve,
+          chainId: (request as AddEthereumChainRequest).params.chainId,
+          rpcUrls: (request as AddEthereumChainRequest).params.rpcUrls,
+          blockExplorerUrls: (request as AddEthereumChainRequest).params.blockExplorerUrls,
+          chainName: (request as AddEthereumChainRequest).params.chainName,
+          iconUrls: (request as AddEthereumChainRequest).params.iconUrls,
+          nativeCurrency: (request as AddEthereumChainRequest).params.nativeCurrency,
+        })
+        // TODO: handle not inline situation
```

Comment:
@fan-zhang-sv you need to implement this todo so that the snackbar in the top right hand corner shows up for mobile. see lines 653-656 below